genet/use/cli.py

```diff
@@ -0,0 +1,951 @@
+import json
+import logging
+import os
+import time
+from pathlib import Path
+from typing import Any, Optional
+
+import geopandas as gpd
+import pandas as pd
+from pyproj import CRS
+
+import genet as gn
+import genet.output.sanitiser as sanitiser
+from genet.utils.persistence import ensure_dir
+from genet.variables import EPSG4326
+
+
+def _to_json(dict_to_save: dict, filepath: Path) -> None:
+    with filepath.open("w", encoding="utf-8") as f:
+        json.dump(
+            sanitiser.sanitise_dictionary(dict_to_save), f, ensure_ascii=False, indent=4
+        )
+
+
+def _write_scaled_vehicles(schedule, list_of_scales, output_dir):
+    for i in list_of_scales:
+        scale = float(i) / 100
+        schedule.scale_vehicle_capacity(scale, scale, output_dir)
+
+
+def _generate_validation_reports(network, output_dir: Path) -> None:
+    logging.info("Generating validation report")
+    report = network.generate_validation_report()
+    logging.info(f'Graph validation: {report["graph"]["graph_connectivity"]}')
+    if network.schedule:
+        logging.info(
+            f'Schedule level validation: {report["schedule"]["schedule_level"]["is_valid_schedule"]}'
+        )
+        if "vehicle level" in report["schedule"]:
+            logging.info(
+                f'Schedule vehicle level validation: {report["schedule"]["vehicle_level"]["vehicle_definitions_valid"]}'
+            )
+        logging.info(
+            f'Routing validation: {report["routing"]["services_have_routes_in_the_graph"]}'
+        )
+
+    _to_json(report, output_dir / "validation_report.json")
+
+    logging.info("Generating summary report")
+    summary_report = network.summary_report()
```

Comment:
This is the biggest change to code, summary report is very useful and goes hand in hand with validation of the network